src/ctapipe_io_lst/__init__.py

```diff
@@ -652,26 +651,36 @@ def _generator(self):
                 self.fill_pointing_info(array_event)
 
             # apply low level corrections
-            if self.apply_drs4_corrections:
-                self.r0_r1_calibrator.apply_drs4_corrections(array_event)
+            tdp_action = array_event.lst.tel[self.tel_id].evt.tdp_action
+            is_calibrated = False
+            if tdp_action is not None:
+                tdp_action = EVBPreprocessingFlag(int(tdp_action))
+                is_calibrated = EVBPreprocessingFlag.PE_CALIBRATION in tdp_action
 
+            if self.apply_drs4_corrections and not is_calibrated:
```

Comment:
is_calibrated refers to the conversion to p.e., right?
If apply_drs4_correction is True, shouldn't it always be the case that is_calibrated = False?

Reply:
`apply_drs4_corrections` is a user given option, `is_calibrated` is determined from the actual data.

So the logic here is mainly to simplify life for analyzers / osa that you can keep request drs4 calibration and give the drs4 calibration file, but the source will figure out if it is needed.